packages/blade/src/components/Checkbox/CheckboxInput.web.tsx

```diff
@@ -0,0 +1,68 @@
+/* eslint-disable @typescript-eslint/ban-types */
+import type { CSSObject } from 'styled-components';
+import styled from 'styled-components';
+import type { Theme } from '../BladeProvider';
+import type { CheckboxIconProps } from './CheckboxIcon';
+import type { InputProps } from './useCheckbox';
+import { getIn, makeMotionTime } from '~utils';
+import { screenReaderStyles } from '~components/VisuallyHidden';
+
+type HoverProps = Pick<CheckboxIconProps, 'isNegative' | 'isDisabled'> & {
+  isChecked: boolean;
+};
+
+const variants = {
+  default: {
+    background: {
+      checked: 'colors.brand.primary.600',
+      unchecked: 'colors.brand.gray.400',
+    },
+  },
+};
+
+const getHoverStyles = ({
+  theme,
+  isDisabled,
+  isNegative,
+  isChecked,
+}: { theme: Theme } & HoverProps): CSSObject => {
+  if (isDisabled || isNegative) return {};
+
+  const checked = isChecked ? 'checked' : 'unchecked';
+  const backgroundColor = variants.default.background[checked];
+
+  return {
+    backgroundColor: getIn(theme, backgroundColor),
+    transitionTimingFunction: theme.motion.easing.standard.effective as string,
+    transitionDuration: makeMotionTime(theme.motion.duration['2xquick']),
+  };
+};
+
+const StyledInput = styled.input<HoverProps>(({ theme, isChecked, isDisabled, isNegative }) => ({
+  ...screenReaderStyles,
+  '&:focus + div': {
+    outline: '1px solid white',
```

Comment:
Shouldn't this value `white` come from theme? I suppose it wouldn't be a white outline always?

Reply:
Yes it will always be white. Tho I agree we can use token here.

Reply:
Tho we do not have a token for focus outline in the theme right now. We need to revise it from design side first.

Reply:
Can we make a note of this? Tokens for outline?